### ISSUE TITLE
[Feature]: <should never turn into a transfer-disabled store.>

### ISSUE THREAD

**Issue by ryujimorimoto** [Shopify/cli#693]
### What area(s) will this request affect?

App, Theme, Running your code locally, Other

### What type of change do you want to see?

Substantial change to existing feature

### Overview

I don't think Shopify CLI 3 should turn into a transfer-disabled store.
Custom app stores launched using the CLI should be limited to transfer-disabled stores only.
I believe that CLI should not be used to let merchants choose transferable development stores

### Motivation

I've run into a problem with custom app development using Shopify CLI 3.
It looks like you accidentally changed production development store of client to a transfer-disabled store when I used the CLI command `yarn dev --reset` and selected client store. I ended up with a store that can't be transferd to a merchant.

The official documentation in the attachment states that installing a custom app will void the transfer.
But actually just selecting a store in the CLI seems to convert it to a transfer-disabled store.
Also, no confirmation message was displayed when converting.
It just displays a message that the conversion was successful.

I don't think the CLI should be converted to a development store.
This is because you may make the wrong choice.

I lost my client's trust by choosing the wrong store.

**Comment by knynkwl** [comment]
Agreed, this was very unclear to me as well and disabled a store that we have been working on for months. This should be more clear in the docs or CLI.
